### PR TITLE
Extend `spicy_add_analyzer` to allow installing additional Zeek scripts.

### LINKED ISSUE #60: CMake function `spicy_add_analyzer` should support installing additional scripts
We provide `spicy_add_analyzer` to compile and install analyzers. We should add support to that script so accompanying Zeek scripts can be installed.

 See e.g., zeek/spicy-analyzers#77 for a related issue.

### PR DESCRIPTION
**Pull request by bbannier**
This patch changes `spicy_add_analyzer` to also support specifying
scripts which should be installed with the plugin. For that we change
the function to now support named arguments (`NAME`, `SOURCES`, and
`SCRIPTS`). For now we continue to support the previous format where
arguments were all unnamed, however this form does not support
installing scripts.

Closes #60.

### CODE REVIEW

File: cmake/ZeekSpicyAnalyzerSupport.cmake [hunk 1]
```diff
@@ -6,32 +6,65 @@
...
+function(spicy_add_analyzer)
+    set(options)
+    set(oneValueArgs NAME)
+    set(multiValueArgs SOURCES SCRIPTS)
+
+    cmake_parse_arguments(PARSE_ARGV 0 SPICY_ANALYZER "${options}" "${oneValueArgs}" "${multiValueArgs}")
+
+    # We also support the legacy behavior where the first arg is
+    # the analyzer NAME and all remaining arguments are SOURCES.
+    if (SPICY_ANALYZER_UNPARSED_ARGUMENTS)
```
Comment (rsmmr) [inline cmake/ZeekSpicyAnalyzerSupport.cmake:27]: Formatting nit:

```suggestion
    if ( SPICY_ANALYZER_UNPARSED_ARGUMENTS )
```

Similar below.

File: cmake/ZeekSpicyAnalyzerSupport.cmake [hunk 2]
```diff
@@ -6,32 +6,65 @@
...
+    add_custom_target(${SPICY_ANALYZER_NAME} ALL DEPENDS ${OUTPUT})
 
     if ( SPICY_MODULE_OUTPUT_DIR_INSTALL )
-        install(FILES ${output} DESTINATION "${SPICY_MODULE_OUTPUT_DIR_INSTALL}")
+        install(FILES ${OUTPUT} DESTINATION "${SPICY_MODULE_OUTPUT_DIR_INSTALL}")
+
+        if (DEFINED SPICY_ANALYZER_SCRIPTS)
+            install(
+                FILES ${SPICY_ANALYZER_SCRIPTS}
+                DESTINATION "${SPICY_SCRIPTS_OUTPUT_DIR_INSTALL}/spicy-analyzers/${NAME}")
```
Comment (rsmmr) [inline cmake/ZeekSpicyAnalyzerSupport.cmake:62]: This won't quite work yet: In the future we want to use this also for packages other than `spicy-analyzers`, so we shouldn't hardcode the name. Not immediately sure though what the solution is. Do we need another argument `PACKAGE_NAME` for this?

Comment (bbannier) [reply cmake/ZeekSpicyAnalyzerSupport.cmake]: Good catch. I went with an explicit `PACKAGE_NAME` argument which becomes required if `SCRIPTS` is also given.

File: cmake/ZeekSpicyAnalyzerSupport.cmake [hunk 3]
```diff
@@ -6,32 +6,69 @@
...
-    string(TOLOWER "${name}" name_lower)
-    set(output "${SPICY_MODULE_OUTPUT_DIR_BUILD}/${name_lower}.hlto")
-    set(output_install "${SPICY_MODULE_OUTPUT_DIR_INSTALL}/${name_lower}.hlto")
-    set(deps "spicyz")
+# Add target to build an analyzer.
+#
+# Usage:
+#
+#     spicy_add_analyzer(
+#         NAME <name>
```
Comment (rsmmr) [inline cmake/ZeekSpicyAnalyzerSupport.cmake:14]:
```suggestion
#         NAME <analyzer_name>
```

**Review by rsmmr — APPROVED** [review]

**Pull request merged**